src/maintagent/agent/process.c

```diff
@@ -449,3 +450,77 @@ FUNCTION void removeOrphanedRows()
 
   return;  // success
 }
+
+
+/**
+ * Remove orphan log files created to store the logs from agents on disk.
+ * @returns void but writes status to stdout
+ */
+FUNCTION void removeOrphanedLogFiles()
+{
+  PGresult* result;
+  PGresult* updateResult;
+  int row;
+  int countTuples;
+  int removedCount = 0;
+  int jobQueueId;
+  char* logPath;
+  long startTime, endTime;
+  fo_dbManager_PreparedStatement* updateStatement;
+  struct stat statbuf;
+
+  char *SQL = "SELECT jq_pk, jq_log FROM job ja "
+    "INNER JOIN job jb ON ja.job_upload_fk = jb.job_upload_fk "
```

Comment:
Joining same table ?? i think the below query is enough.

```
SELECT jq_pk, jq_log FROM job ja 
    INNER JOIN jobqueue jq ON ja.job_pk = jq.jq_job_fk 
    WHERE ja.job_name = 'Delete' AND jq_log IS NOT NULL 
    AND jq_log != 'removed';
```

Reply:
This was done as the `Delete` job as a separate entry in `job_queue` and `job` table.

The same table was joined back on `upload_fk` to get all the jobs run on the uploads with `delete` jobs.